### PR TITLE
Upgrade handler code-gen

### PR DESCRIPTION
**Pull request by MasterPi-2124**
Closes: #3632 and #3628

## Testing and Verifying

I created a pr on notional-labs/osmosis, which can be seen here https://github.com/notional-labs/osmosis/pull/96

### CODE REVIEW

**Comment by MasterPi-2124** [comment]
@p0mvn can you take a look into that? Thankyou <3

File: scripts/empty_upgrade_handler_gen.sh [hunk 1]
```diff
@@ -81,15 +82,40 @@ func CreateUpgradeHandler(
...
+NEW_IMPORT="$version_create $module/app/upgrades/$version_create$bracks"
+sed -i "s|.*$PREV_IMPORT.*|\t$PREV_IMPORT\n\t$NEW_IMPORT|" $app_file
+
+# change e2e version in makefile
+sed -i "s/E2E_UPGRADE_VERSION := ${bracks}v$latest_version$bracks/E2E_UPGRADE_VERSION := ${bracks}$version_create$bracks/" ./Makefile
+
+# bumps up prev e2e version
+e2e_file=./tests/e2e/containers/config.go
+PREV_OSMOSIS_DEV_TAG=$(curl -L -s 'https://registry.hub.docker.com/v2/repositories/osmolabs/osmosis-dev/tags?page=1&page_size=100'            | jq -r '.results[] | .name | select(.|test("^(?:v|)[0-9]+\\.[0-9]+(?:$|\\.[0-9]+$)"))' | grep -m1 "")
+PREV_OSMOSIS_E2E_TAG=$(curl -L -s 'https://registry.hub.docker.com/v2/repositories/osmolabs/osmosis-e2e-init-chain/tags?page=1&page_size=100' | jq -r '.results[] | .name | select(.|test("^(?:v|)[0-9]+\\.[0-9]+(?:$|\\.[0-9]+$)"))' | grep -m1 "")
```
Comment (RusAkh) [inline scripts/empty_upgrade_handler_gen.sh:101]: Hey @MasterPi-2124! Thanks for the PR. I might be missing something, but shouldn't it be like this? Not a pro in grep flags so correct me if I am wrong :)
```suggestion
PREV_OSMOSIS_E2E_TAG=$(curl -L -s 'https://registry.hub.docker.com/v2/repositories/osmolabs/osmosis-e2e-init-chain/tags?page=1&page_size=100' | jq -r '.results[] | .name | select(.|test("^(?:v|)[0-9]+\\.[0-9]+(?:$|\\.[0-9]+$)"))' | grep -m 1 "")
```

**Review by RusAkh — APPROVED** [review]
LGTM!

File: scripts/empty_upgrade_handler_gen.sh (hunk 1 above)
Comment (MasterPi-2124) [reply scripts/empty_upgrade_handler_gen.sh:101]: actually both are correct, but for clarity for other people, I will change that into the full flag

Comment (RusAkh) [reply scripts/empty_upgrade_handler_gen.sh:101]: got it! thanks for the clarification

**Comment by p0mvn** [comment]
@MasterPi-2124 Thanks for the change. Please apply the state compatibility labels

File: .github/workflows/gen-upgrade-on-manual-input.yml [hunk 2]
```diff
@@ -45,14 +45,14 @@ jobs:
...
-        if: env.MAJOR == 1    # 1 means vX of existing upgrade handler is smaller than A in tag vA.B.C
+        if: env.MAJOR == 1 # 1 means vX of existing upgrade handler is smaller than A in tag vA.B.C
         run: bash ./scripts/empty_upgrade_handler_gen.sh ${{ env.input }}
 
       - name: Create PR
         if: env.MAJOR == 1
         uses: peter-evans/create-pull-request@v4
         with:
-          base: ${{ github.event.repository.default_branch }}
+          token: ${{ secrets.COMMIT_TO_BRANCH }}
```
[the remark this answers is not in the document]
Comment (p0mvn) [reply .github/workflows/gen-upgrade-on-manual-input.yml:55]: Ahh it seems to me that we need this for the similar reasons to the import paths action: https://github.com/osmosis-labs/osmosis/blob/258ee1c00cd2464a278bef996d9b6aec3ad1e803/.github/workflows/import_paths.yml#L65

**Review by p0mvn — APPROVED** [review]
LGTM, thank you, going to test shortly

**Pull request merged**